packages/eslint-plugin/src/rules/no-invalid-void-type.ts

```diff
@@ -127,6 +127,35 @@ export default util.createRule<[Options], MessageIds>({
       }
     }
 
+    /**
+     * @brief checks if the generic type parameter extends void
+     */
+    function checkExtendedVoid(
+      node: TSESTree.TSVoidKeyword,
+      parentNode: TSESTree.TSTypeParameter,
+    ): void {
+      if (
+        parentNode.constraint?.type !== AST_NODE_TYPES.TSVoidKeyword ||
```

Comment:
I'm confused, why do we care about `parentNode.constraint`? Shouldn't we exclusively care about `parentNode.default`?

Reply:
Sorry to confuse you, that's because of my lack of understanding...
I deleted `parentNode.constraint` from if statement.